packages/core/src/dispatcher.ts

```diff
@@ -151,6 +151,10 @@ export class Dispatcher {
           })
         }
 
+        // put the payload into ipfs
+        jws.link = await this._shutdownSignal.abortable((signal) => {
```

Comment:
yeah I actually feel like overriding the `jws.link` here with what we get back obfuscates the behavior a bit.  It's now less obvious that we expect the CID not to change because we are putting a fully constructed block.  I'd prefer to either put things back how they were or add a check that the CID didn't change so it's clearer what the behavioral expectations are.

Reply:
Will add a check that cid did not change.